tools/windows-node-installer/test/e2e/clustersetup.go

```diff
@@ -0,0 +1,195 @@
+			_, err := core.ServiceAccounts(projectName).Get(resourceName, metav1.GetOptions{})
+			if err != nil {
+				_, err := core.ServiceAccounts(projectName).Create(o)
+				if err != nil {
+					return fmt.Errorf("error creating  role: %v", err)
+				}
+				log.Print("Service Account Created")
+			}
+
+		case *rbac.RoleBinding:
+			// check if the role binding already exists
+			_, err := rbacConfig.RoleBindings(projectName).Get(resourceName, metav1.GetOptions{})
+			if err != nil {
+				_, err := rbacConfig.RoleBindings(projectName).Create(o)
+				if err != nil {
+					return fmt.Errorf("error creating  role binding %v", err)
+				}
+				log.Print("Role Binding created")
+			}
+		case *rbac.ClusterRoleBinding:
+			// check if the cluster role binding already exists
+			_, err := rbacConfig.ClusterRoleBindings().Get(resourceName, metav1.GetOptions{})
+			if err != nil {
+				_, err := rbacConfig.ClusterRoleBindings().Create(o)
+				if err != nil {
+					return fmt.Errorf("error creating  cluster role binding %v", err)
+				}
+				log.Print("Cluster Role Binding created")
+			}
+		default:
+			return fmt.Errorf("invalid resource file")
+
+		}
+	}
+	return nil
+}
+
+// getServiceAccountToken sets the values for serviceAccountToken variable.
+func getServiceAccountToken(cfg *rest.Config) (string, error) {
+	core, err := coreclient.NewForConfig(cfg)
+	if err != nil {
+		return "", fmt.Errorf("error getting config for core resources: %v", err)
+	}
+	// get service account
+	sa, err := core.ServiceAccounts(projectName).Get(resourceName, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("error getting the service account : %v", err)
+	}
+	var tokenSecret = ""
+	// get secret token for the service account
+	for _, secret := range sa.Secrets {
+		if strings.Contains(secret.Name, resourceName+"-token") {
+			tokenSecret = secret.Name
+		}
+	}
+	// get secret object using secret name value
+	secret, err := core.Secrets(projectName).Get(tokenSecret, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("error getting secret for the service account %v", err)
+	}
+	serviceAccountTokenBytes := secret.Data["token"]
+	// decode service account token
+	saToken, err := base64.StdEncoding.DecodeString(base64.StdEncoding.EncodeToString(serviceAccountTokenBytes))
+	if err != nil {
+		return "", fmt.Errorf("error while decoding the service account token %v", err)
+	}
+	return string(saToken), nil
+}
+
+// getOpenShiftClientForSA returns config specific to the service account created in the test cluster. It uses service
+// account token to update existing config to use service account permissions.
+func getOpenShiftClientForSA(kubeconfig string) (*client.OpenShift, error) {
+	cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("error creating config object %v", err)
+	}
+	// Create roles, role bindings and service account
+	if err := setRbacConfig(cfg); err != nil {
```

Comment:
We should clean up the stuff created when we are done with it. SA, roles, etc.

Reply:
Done